### PR TITLE
translate-c: demote usage of un-implemented builtins

### PR DESCRIPTION
**Pull request by ehaas**

### CODE REVIEW

File: src/translate_c.zig [hunk 1]
```diff
@@ -1538,6 +1539,22 @@ fn transImplicitCastExpr(
     }
 }
 
+fn isBuiltinDefined(name: []const u8) bool {
+    inline for (std.meta.declarations(c_builtins)) |decl| {
```
Comment (daurnimator) [inline src/translate_c.zig:1543]: Use `@hasDecl` to avoid the loop.

**Review by daurnimator — DISMISSED** [review]

File: src/translate_c.zig (hunk 1 above)
Comment (ehaas) [reply src/translate_c.zig:1543]: Doesn't that require the name to be comptime known?

Comment (daurnimator) [reply src/translate_c.zig:1543]: ah right; I didn't comprehend that this needed to work at runtime.

File: src/translate_c.zig [hunk 2]
```diff
@@ -1538,6 +1539,22 @@ fn transImplicitCastExpr(
...
+        if (std.mem.eql(u8, name, decl.name)) return true;
+    }
+    return false;
+}
+
+fn transBuiltinFnExpr(c: *Context, scope: *Scope, expr: *const clang.Expr, used: ResultUsed) TransError!Node {
+    const node = try transExpr(c, scope, expr, used);
+    if (node.castTag(.identifier)) |ident| {
+        const name = ident.data;
+        if (!isBuiltinDefined(name)) return fail(c, error.UnsupportedTranslation, expr.getBeginLoc(), "TODO implement function '{s}' in std.c.builtins", .{name});
```
Comment (Vexu) [inline src/translate_c.zig:1553]: Could you also add this check to `parseCPrimaryExprInner` for consistency?

**Review by Vexu — APPROVED** [review]

File: src/translate_c.zig (hunk 2 above)
Comment (ehaas) [reply src/translate_c.zig:1553]: Updated, although I'm not sure if I should be using `mangled_name` or `slice` there.

Comment (Vexu) [reply src/translate_c.zig:1553]: `mangled_name` should be fine.

**Pull request merged**